hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -515,8 +515,11 @@ private ContainerCommandResponseProto runCommand(
 
         ContainerCommandResponseProto response = runCommand(
             requestProto, context);
-        String path = response.getMessage();
-        return new LocalStream(new StreamDataChannel(Paths.get(path)));
+        final StreamDataChannel channel = new StreamDataChannel(
+            Paths.get(response.getMessage()));
+        final ExecutorService chunkExecutor = requestProto.hasWriteChunk() ?
+            getChunkExecutor(requestProto.getWriteChunk()) : null;
+        return new LocalStream(channel, chunkExecutor);
```

Comment:
Hi @szetszwo, The size of the chunkExecutor is determined by dfs.container.ratis.num.write.chunk.threads.per.volume.  So I'm wondering if we need delete  [datastream.write.threads in DatanodeRatisServerConfig](https://github.com/apache/ozone/blob/HDDS-4454/hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/conf/DatanodeRatisServerConfig.java#L144)? Now that we're passing in chunkExecutor every time, this configuration should be useless.

Reply:
@captainzmc , you are right.  Let's remove it.